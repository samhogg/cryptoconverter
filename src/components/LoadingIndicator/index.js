import React from "react";
import PropTypes from "prop-types";
import styled, { keyframes } from "styled-components";

const pulse = keyframes`
  0% {
    transform: translate3d(0);
    opacity: 0.4;
  }
  50% {
    transform: translate3d(0, -6px, 0);    
    opacity: 1
  }
  100% {
    transform: translate3d(0);
    opacity: 0.4;
  }
`;

const LoadingContainer = styled.div`
  position: absolute;
  right: 1.2rem;
  top: 1.2rem;
  transition: opacity 0.3s;
  opacity: ${({ loading }) => (loading ? 1 : 0)};
`;

const LoadingDot = styled.div`
  display: inline-block;
  border-radius: 50%;
  margin-right: 3px;
  width: 6px;
  height: 6px;
  background: ${({ theme }) => theme.colors.primary};
  animation: ${pulse} 1s linear infinite;
  &:nth-child(2) {
    animation-delay: 0.333s;
  }
  &:nth-child(3) {
    animation-delay: 0.666s;
  }
`;

const LoadingIndicator = ({ loading }) => (
  <LoadingContainer loading={!!loading}>
    <LoadingDot />
    <LoadingDot />
    <LoadingDot />
  </LoadingContainer>
);

LoadingIndicator.propTypes = {
  loading: PropTypes.bool
};

export default LoadingIndicator;
