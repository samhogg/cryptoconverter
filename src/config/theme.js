const theme = {
  colors: {
    primary: "#0062AE",
    error: "#be0000"
  },

  // System Font Stack
  // Adapted from https://github.com/mrmrs/css-system-fonts
  fontStack: `-apple-system, BlinkMacSystemFont,
    /* MacOS and iOS */ "avenir next", avenir, /* MacOS and iOS */ "Segoe UI",
    /* Windows */ "lucida grande", /* Older MacOS */ "helvetica neue", helvetica,
    /* Older MacOS */ "Fira Sans", /* Firefox OS */ roboto, noto,
    /* Google stuff */ "Droid Sans", /* Old Google stuff */ cantarell, oxygen,
    ubuntu, /* Linux stuff */ "franklin gothic medium", "century gothic",
    /* Windows stuff */ "Liberation Sans", /* Linux */ sans-serif; /* Everything else */`
};

export default theme;
