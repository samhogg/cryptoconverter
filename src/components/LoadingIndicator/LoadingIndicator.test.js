import React from "react";
import LoadingIndicator from ".";
import renderer from "react-test-renderer";
import { ThemeProvider } from "styled-components";
import theme from "../../config/theme";

it("Renders with loading=true", () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={theme}>
        <LoadingIndicator loading={true} />
      </ThemeProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("Renders with loading=false", () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={theme}>
        <LoadingIndicator loading={false} />
      </ThemeProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
