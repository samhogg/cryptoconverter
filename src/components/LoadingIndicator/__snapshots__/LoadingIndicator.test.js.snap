// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders with loading=false 1`] = `
<div
  className="LoadingIndicator__LoadingContainer-eMCbod fjLIAU"
>
  <div
    className="LoadingIndicator__LoadingDot-kbVGSy eVeAmw"
  />
  <div
    className="LoadingIndicator__LoadingDot-kbVGSy eVeAmw"
  />
  <div
    className="LoadingIndicator__LoadingDot-kbVGSy eVeAmw"
  />
</div>
`;

exports[`Renders with loading=true 1`] = `
<div
  className="LoadingIndicator__LoadingContainer-eMCbod bZALBS"
>
  <div
    className="LoadingIndicator__LoadingDot-kbVGSy eVeAmw"
  />
  <div
    className="LoadingIndicator__LoadingDot-kbVGSy eVeAmw"
  />
  <div
    className="LoadingIndicator__LoadingDot-kbVGSy eVeAmw"
  />
</div>
`;
